workflow/controller/workflowpod.go

```diff
@@ -503,13 +508,18 @@ func (woc *wfOperationCtx) createWorkflowPod(ctx context.Context, nodeName strin
 		return nil, err
 	}
 
-	if templateDeadline != nil && (pod.Spec.ActiveDeadlineSeconds == nil || time.Since(*templateDeadline).Seconds() < float64(*pod.Spec.ActiveDeadlineSeconds)) {
-		newActiveDeadlineSeconds := int64(time.Until(*templateDeadline).Seconds())
-		if newActiveDeadlineSeconds <= 1 {
-			return nil, fmt.Errorf("%s exceeded its deadline", nodeName)
+	if templateDeadline != nil {
+		if !pendingOnly && (pod.Spec.ActiveDeadlineSeconds == nil || time.Since(*templateDeadline).Seconds() < float64(*pod.Spec.ActiveDeadlineSeconds)) {
+			newActiveDeadlineSeconds := int64(time.Until(*templateDeadline).Seconds())
+			if newActiveDeadlineSeconds <= 1 {
```

Comment:
It seems slightly odd that we cut off with 1 second remaining on the deadline - any reason this is not zero?